ui/apps/platform/src/Containers/ComplianceEnhanced/Schedules/hooks/useWatchLastSnapshotForComplianceReports.ts

```diff
@@ -0,0 +1,90 @@
+import { useCallback } from 'react';
+
+import useInterval from 'hooks/useInterval';
+import { getRequestQueryStringForSearchFilter } from 'utils/searchUtils';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+import {
+    ComplianceReportSnapshot,
+    ComplianceScanConfigurationStatus,
+    fetchComplianceReportHistory,
+} from 'services/ComplianceScanConfigurationService';
+import useRestQuery from 'hooks/useRestQuery';
+
+async function fetchLastJobForConfiguration(
+    scanConfigId: string
+): Promise<ComplianceReportSnapshot | null> {
+    const query = getRequestQueryStringForSearchFilter({
+        'Report state': ['PREPARING', 'WAITING'],
+    });
+
+    const complianceReportSnapshots = await fetchComplianceReportHistory({
+        id: scanConfigId,
+        query,
+        page: 1,
+        perPage: 1,
+        showMyHistory: true,
+        sortOption: {
+            field: 'Report Completion Time',
+            reversed: true,
+        },
+    });
+
+    return complianceReportSnapshots[0] ?? null;
+}
+
+type ComplianceReportSnapshotLookup = Record<string, ComplianceReportSnapshot | null>;
+
+type Result = {
+    complianceReportSnapshots: ComplianceReportSnapshotLookup;
+    isLoading: boolean;
+    error: string | null;
+};
+
+export type FetchLastComplianceReportSnapshotReturn = Result & {
+    fetchSnapshots: () => void;
+};
+
+function useWatchLastSnapshotForComplianceReports(
+    scanConfigurations: ComplianceScanConfigurationStatus[] | undefined
+): FetchLastComplianceReportSnapshotReturn {
+    const fetchSnapshotsCallback = useCallback(async () => {
+        if (!scanConfigurations) {
+            const result: ComplianceReportSnapshotLookup = {};
+            return Promise.resolve(result);
+        }
+
+        const promise: Promise<ComplianceReportSnapshotLookup> = new Promise((resolve, reject) => {
+            const configurations = Array.isArray(scanConfigurations)
+                ? scanConfigurations
+                : [scanConfigurations];
+
+            Promise.all(configurations.map(({ id }) => fetchLastJobForConfiguration(id)))
+                .then((snapshotResults) => {
+                    const result: ComplianceReportSnapshotLookup = configurations.reduce(
+                        (acc, { id }, index) => ({ ...acc, [id]: snapshotResults[index] }),
+                        {}
+                    );
+                    resolve(result);
+                })
+                .catch((error) => {
+                    reject(error);
```

Comment:
should we propagate an error if it fails?

Reply:
I would assume so. Do you feel differently?

Reply:
oh no i meant in the UI to surface to the user

Reply:
That's a good point. We don't do that in the VM Report workflow either. Right now, If there is an error we ignore it in the UI but still pass the error to the backend for logging purposes. I can talk to Mansur about this a bit more.